### ISSUE TITLE
[Improvement] track open state of "purchased unit" instead of "stock unit"

### ISSUE THREAD

**Issue by luclu** [grocy/grocy#729]
Consider the following:
- one product with 
  -  `Quantity unit purchase` set to `Pack`
  -  `Quantity unit stock` set to `Gramm`

In the stock overview pressing the `mark as open` button should not mark `1 Gramm` as open, but the whole `pack`.

**Comment by berrnd** [comment]
Currently all "actions" are related (only) to the products stock quantity unit.

Duplicate of #711 (will be merged into this).

**Comment by berrnd** [comment]
_Another example by @Eyeball753 originally posted in #711:_

> I'm just gonna lead with an example to explain it better.
I buy my butter in a pack of 4 sticks. I add it as grams instead of sticks and 1 stick is 113 grams so a pack is 452 grams of butter.
My idea is to be able to tell the butter product that 1 stick is 113g so when I want to consume 1 stick it'll consume 113g. But also when I've used a certain amount of butter already, for example 13g, that if I tell it to consume 1 stick, it would consume the remainder of what's left of that stick, the 100g. So 339g (total of 3 sticks) would be left.
I understand that choosing a unit to consume based on the conversion rates we entered (the product is in gram, but at consume we select kg or any unit if it is converted) is already a request, but when that gets added and I were to convert 1 stick to 113g gram, it would just always remove 113g of butter, and not the remainder of what was left of that stick.
